examples/twittermap/web/public/javascripts/map/countmap.js

```diff
@@ -58,60 +58,124 @@ angular.module('cloudberry.map')
 
     // Send query to cloudberry
     function sendCountmapQuery() {
-      // Batch request without map result - used when the complete map result cache hit case
+      // For time-series histogram, get geoIds not in the time series cache.
+      $scope.geoIdsNotInTimeSeriesCache = TimeSeriesCache.getGeoIdsNotInCache(cloudberry.parameters.keywords,
+        cloudberry.parameters.timeInterval, cloudberry.parameters.geoIds, cloudberry.parameters.geoLevel);
+
+      // Batch request without map result - used when the complete map result cache hit, partial time series result cache hit case
       var batchWithoutGeoRequest = cloudberryConfig.querySliceMills > 0 ? {
-        batch: [queryUtil.byTimeRequest(cloudberry.parameters)],
+        batch: [queryUtil.byTimeRequest(cloudberry.parameters, $scope.geoIdsNotInTimeSeriesCache)],
         option: {
           sliceMillis: cloudberryConfig.querySliceMills
         }
       } : {
-        batch: [queryUtil.byTimeRequest(cloudberry.parameters)]
+        batch: [queryUtil.byTimeRequest(cloudberry.parameters, $scope.geoIdsNotInTimeSeriesCache)]
       };
 
       // Gets the Geo IDs that are not in the map result cache.
       $scope.geoIdsNotInCache = MapResultCache.getGeoIdsNotInCache(cloudberry.parameters.keywords,
         cloudberry.parameters.timeInterval,
         cloudberry.parameters.geoIds, cloudberry.parameters.geoLevel);
 
-      // Batch request with only the geoIds whose map result are not cached yet - partial map result cache hit case
+      // Batch request without time series result - used when the complete time series cache hit, partial map result cache hit case
+      var batchWithoutTimeRequest = {
+        batch: [queryUtil.byGeoRequest(cloudberry.parameters, $scope.geoIdsNotInCache)]
```

Comment:
Why this query has no "slicing" check, like batchWithPartialGeoRequest?

Reply:
This query is used with the "complete time series cache hit" case, since the time series results are obtained from the cache immediately (no slice), we can get the corresponding final map result.

Reply:
Oh, I see. In this case, should the time-series histogram displayed in slices? (to be synced with this request)

Reply:
Yes. I think that's what we decided.